source/adapters/cuda/kernel.hpp

```diff
@@ -109,14 +112,21 @@ struct ur_kernel_handle_t_ {
         AlignedLocalMemSize.resize(Index + 1);
         OriginalLocalMemSize.resize(Index + 1);
       }
-      ParamSizes[Index] = Size;
-      // calculate the insertion point on the array
-      size_t InsertPos = std::accumulate(std::begin(ParamSizes),
-                                         std::begin(ParamSizes) + Index, 0);
-      // Update the stored value for the argument
-      std::memcpy(&Storage[InsertPos], Arg, Size);
-      Indices[Index] = &Storage[InsertPos];
-      AlignedLocalMemSize[Index] = LocalSize;
+
+      // Copy new argument to storage if it hasn't been added before.
+      if (ParamSizes[Index] == 0) {
+        ParamSizes[Index] = Size;
+        std::memcpy(&Storage[InsertPos], Arg, Size);
+        Indices[Index] = &Storage[InsertPos];
```

Comment:
This isn't an issue introduced by this PR, but the description of `Indices` as `Byte offset into /p Storage allocation for each parameter.` is a little confusing, isn't it? An index, to me, implies a relative value from the beginning of the storage, not an absolute address. Do you think we should do something about this? I note this PR is embedding the use of `Indices` more deeply into the code so it might be best to fix now.

Reply:
Yes, I don't mind changing the variable name and related getters. Do you have a preference for the name? Perhaps `ParamPointers` or `ParamOffsets` `?

Reply:
The existing getter was called `getArgIndices()` so I decided to rename the variable to `ArgPointers`.  Hopefully that name makes more sense.

Reply:
Sounds good to me, thank you